### ISSUE TITLE
[Bug] markline may not be centered at different phone resolutions

### ISSUE THREAD

**Issue by rookie-hhm** [apache/echarts#18509]
### Version

5.3.2

### Link to Minimal Reproduction

https://codesandbox.io/s/adoring-sea-psc4st?file=/src/index.js

### Steps to Reproduce

1. open [link](https://codesandbox.io/s/adoring-sea-psc4st?file=/package.json:249-254)， change Viewport width
2. you can see markline move and it's not centered at some widths

### Current Behavior

markline doesn't stay centered in bar chart
![image](https://user-images.githubusercontent.com/34794050/231701295-f0d660c8-057b-4799-950c-008175a967dd.png)

### Expected Behavior

markline can always be centered

### Environment

```markdown
- OS: macOS Monterey
- Browser: Chrome 112.0.5615.49
- Framework: vue2
```

**Comment by Ovilia** [comment]
<img width="173" alt="image" src="https://user-images.githubusercontent.com/779050/232658292-bb49890c-d6db-404e-b533-ec493aff4f48.png">

Looks good to me. Can you make the demo with the width that can reproduce the problem?

**Comment by rookie-hhm** [comment]
> <img alt="image" width="173" src="https://user-images.githubusercontent.com/779050/232658292-bb49890c-d6db-404e-b533-ec493aff4f48.png">
> 
> Looks good to me. Can you make the demo with the width that can reproduce the problem?

Yeah,I have set viewport width '941px' . And I add these code to resize chart width. 
![image](https://user-images.githubusercontent.com/34794050/233551841-50486a50-97f7-4810-878d-1ef9ca2064a5.png)
You can adjust the viewport size to see how the markline changes.
Please open this  [link](https://codesandbox.io/s/adoring-sea-psc4st?file=/src/index.js)

**Comment by rookie-hhm** [comment]
I also encountered this problem in v4.9.0, and it feels like an ongoing problem
